internal/broker/broker.go

```diff
@@ -102,6 +111,13 @@ type Option func(*option)
 func New(cfg Config, args ...Option) (b *Broker, err error) {
 	defer decorate.OnError(&err, "could not create broker")
 
+	if cfg.ConfigFile != "" {
+		cfg.userConfig, err = parseConfigFile(cfg.ConfigFile)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse config: %v", err)
+		}
+	}
+
```

Comment:
From its name, I would expect the `parseConfigFile` function to return an error if there is nothing to parse. I think that we want to continue in that case though, so that we don't have to create and parse a config file in the tests. We could achieve that by returning a custom error in `parseConfigFile` and handling that by ignoring it, but the current approach is simpler and achieves the same.